### PR TITLE
Beds 547/dashboard manage validator

### PR DESCRIPTION
**Pull request by marcel-bitfly**

### CODE REVIEW

File: frontend/components/bc/BcInputSearch.vue [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  isLoading: boolean,
+  results: T[] | undefined,
+  shouldClearOnSubmit?: boolean,
+  shouldSelectFirstResult?: boolean,
+}>(), {
+  shouldClearOnSubmit: true,
+})
+
+const emit = defineEmits<{
+  (e: 'search', value: string): void,
```
Comment (benji-bitfly) [inline frontend/components/bc/BcInputSearch.vue]: np: I wonder if this works and make things clearer 🤔 
```suggestion
  (e: 'search', input: string): void,
```

File: frontend/components/bc/BcInputSearch.vue [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  (e: 'submit', value: T): void,
+}>()
+
+const { t: $t } = useTranslation()
+
+const input = defineModel<string>()
+
+const elementInput = ref<HTMLElement >()
```
Comment (benji-bitfly) [inline frontend/components/bc/BcInputSearch.vue:22]: np: (nvm I will research ESLint rule for that) 
```suggestion
const elementInput = ref<HTMLElement>()
```

File: frontend/components/bc/BcInputSearch.vue [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+  currentIndex.value = currentIndex.value + step
+  if (currentIndex.value >= resultItems.value.length) {
+    currentIndex.value = 0
+    return
+  }
+  if (currentIndex.value < 0) {
+    currentIndex.value = resultItems.value.length - 1
+    return
+  }
+}
```
Comment (benji-bitfly) [inline frontend/components/bc/BcInputSearch.vue]: Please split this up into two functions. 
- increaseSelectedIndex()
- decreaseSelectedIndex()

File: frontend/components/bc/BcInputSearch.vue [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+const elementInput = ref<HTMLElement >()
+const hasInput = computed(() => input.value?.length)
+
+const idResults = useId()
+const elementResults = ref<HTMLElement>()
+
+const idElementList = useId()
+const elementList = ref<Array<HTMLElement>>()
+
+const resultItems = computed(() => {
```
Comment (benji-bitfly) [inline frontend/components/bc/BcInputSearch.vue]: Isn't this the same as `results.length`?

File: frontend/components/bc/BcInputSearch.vue [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+})
+</script>
+
+<template>
+  <form
+    role="search"
+    class="bc-input-search__form"
+    @keydown.arrow-up.prevent="hasResults && changeSelectedIndex(-1)"
+    @keydown.arrow-down.prevent="hasResults && changeSelectedIndex(+1)"
+    @keydown.arrow-down.alt.exact.prevent="hasResults && showPopover()"
```
Comment (benji-bitfly) [inline frontend/components/bc/BcInputSearch.vue:148]: Is this really necessary?  
```suggestion
    @keydown.arrow-up="hasResults && changeSelectedIndex(-1)"
    @keydown.arrow-down="hasResults && changeSelectedIndex(+1)"
    @keydown.arrow-down.alt.exact="hasResults && showPopover()"
```

File: frontend/components/bc/BcInputSearch.vue [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+        </div>
+        <slot
+          v-else
+          name="results"
+          :results
+          role="listbox"
+        >
+          <ul
+            role="listbox"
+            aria-label="Add the following validators to your dashbard"
```
Comment (benji-bitfly) [inline frontend/components/bc/BcInputSearch.vue]: Please add `translation`.

Comment (marcel-bitfly) [reply frontend/components/bc/BcInputSearch.vue]: thx 👍

File: frontend/components/bc/BcInputSearch.vue (hunk 4 above)
Comment (marcel-bitfly) [reply frontend/components/bc/BcInputSearch.vue]: true...thx 💯

File: frontend/components/bc/BcInputSearch.vue (hunk 2 above)
Comment (marcel-bitfly) [reply frontend/components/bc/BcInputSearch.vue:22]: kwl thx...so I leave it in for you to see how if it works

File: frontend/components/bc/BcInputSearch.vue (hunk 1 above)
Comment (marcel-bitfly) [reply frontend/components/bc/BcInputSearch.vue]: a little bit yeah...lets do it

**Pull request merged**